src/notification_handling.rs

```diff
@@ -154,36 +149,36 @@ fn handle_payment_received_notification(
     sdk: Arc<BreezServices>,
     event_receiver: Receiver<BreezEvent>,
     payment_hash: String,
-) -> NotificationHandlingResult<RecommendedAction> {
+) -> NotificationHandlingResult<Notification> {
     // Check if the payment was already received
     if let Some(payment) = get_confirmed_payment(&rt, &sdk, &payment_hash)? {
-        return Ok(RecommendedAction::ShowNotification {
-            notification: Notification::Bolt11PaymentReceived {
-                amount_sat: payment.amount_msat / 1000,
-                payment_hash,
-            },
+        return Ok(Notification::Bolt11PaymentReceived {
+            amount_sat: payment.amount_msat / 1000,
+            payment_hash,
         });
     }
 
     // Wait for payment to be received
     if let Some(details) = wait_for_payment_with_timeout(event_receiver, &payment_hash)? {
-        return Ok(RecommendedAction::ShowNotification {
-            notification: Notification::Bolt11PaymentReceived {
-                amount_sat: details.payment.map(|p| p.amount_msat).unwrap_or(0) / 1000, // payment will only be None for corrupted GL payments. This is unlikely, so giving an optional amount seems overkill.
-                payment_hash,
-            },
+        return Ok(Notification::Bolt11PaymentReceived {
+            amount_sat: details.payment.map(|p| p.amount_msat).unwrap_or(0) / 1000, // payment will only be None for corrupted GL payments. This is unlikely, so giving an optional amount seems overkill.
```

Comment:
Yes not an `Option`, but why returning `Ok()` with an `amount_sat` of `0` rather than just returning an error?

What is a _corrupted GL_ payment even? Does it mean the payment did actually go through, but GL cannot provide data for it? Then too I'm not sure whether it is better to display a wrong payment or just no payment at all.

Reply:
> What is a corrupted GL payment even? Does it mean the payment did actually go through, but GL cannot provide data for it?

Yes, the payment goes through.

It's likely better to return information communicating that the expected payment was received, even if the returned amount is incorrect. If we want to capture the possibility of these corrupted payments correctly, an optional value would be best, but I don't think we should add that complexity to the interface to accommodate such an unlikely event.